### ISSUE TITLE
Stop inferring the `PlatformTarget` in the SDK when targeting modern .NET applications

### ISSUE THREAD

**Issue by baronfel** [dotnet/sdk#42558]
Also, you may want to stop inferring the `PlatformTarget` in the SDK when targeting modern .NET applications: 

https://github.com/dotnet/sdk/blob/88c64736ff7039536fc2546d5026d8b78e840e49/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets#L124-L157

_Originally posted by @jkotas in https://github.com/dotnet/sdk/issues/42556#issuecomment-2271913014_

**Comment by marcpopMSFT** [comment]
Triage: The safest solution here is to just add a When after the first one that checks if it's .net core and defaults to AnyCPU. We would need to confirm that cppcli is excluded from that condition. That solution shouldn't require other changes elsewhere.

**Comment by baronfel** [comment]
@joeloff noted that the platform bit in the PE Header may influence registry hive/program files and other defaults - this could be impactful for already-compiled applications.

**Comment by marcpopMSFT** [comment]
Also watch out for what value a solution sets (is it platform rather than platformtarget) and if it's set by the time this code is evaluated.

**Comment by baronfel** [comment]
The Platform<->PlatformTarget relationship is handled by the SDK. The Solution sets `Platform` for the project in the generated metaproj, and the SDK translates that to a PlatformTarget value [here](https://github.com/dotnet/sdk/blob/a7be949662c3cb777935ae5d2751f443f1b72426/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.props#L68-L83). This value then feeds into the logic in RuntimeIdentifierInference afterwards.

**Comment by jkotas** [comment]
> @joeloff noted that the platform bit in the PE Header may influence registry hive/program files and other defaults 

That's true for .NET Framework. I do not think it is the case for .NET Core.
